lib/TableUtils.lua

```diff
@@ -24,6 +24,8 @@ function TableUtils.FlatMap(source, handler) --: ((any[], (element: any, key: nu
 		local list = handler(v, i)
 		if type(list) == "table" then
 			TableUtils.InsertMany(result, list)
+		elseif list then
```

Comment:
hmm, how about we just always insert values, rather than checking for truthiness? if you want to explicitly insert nothing, you just return an empty list...

Reply:
I would prefer that, my only concern was that existing code would be relying on this quirky behaviour. However it appears there is only one call site (in ChapterService) which does not rely on the quirk.